src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+        } yield ()
+      })
+
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  override def getWorkflowState(workflowId: WorkflowId): Future[Option[WorkflowState]] = {
+
+    val action = for {
+
+      workflowExecutionResultOption <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.headOption
+
+      workflowState = workflowExecutionResultOption map { workflowExecutionResult =>
+        WorkflowState.fromString(workflowExecutionResult.status)
+      }
+
+    } yield workflowState
+
+    runTransaction(action)
+  }
+
+  override def getExecutionStatuses(workflowId: WorkflowId): Future[Map[FullyQualifiedName, CallStatus]] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
+
+      executionResults <- dataAccess.executions.filter(symbol =>
+        symbol.workflowExecutionId === workflowExecutionResult.workflowExecutionId).result
+
+      executionStatuses = executionResults.map(executionResult =>
+        executionResult.callFqn -> ExecutionStatus.withName(executionResult.status)
+      ).toMap
+
+    } yield executionStatuses
+
+    runTransaction(action)
+  }
+
+  override def getWorkflowsByState(states: Traversable[WorkflowState]): Future[Traversable[WorkflowInfo]] = {
+
+    val action = for {
+
+      workflowExecutionResults <- dataAccess.workflowExecutions.filter(
+        _.status inSet states.map(_.toString)).result
+
+      workflowInfos <- DBIO.sequence(
+        workflowExecutionResults map { workflowExecutionResult =>
+          val workflowExecutionAuxResult = dataAccess.workflowExecutionAuxes.filter(
+            _.workflowExecutionId === workflowExecutionResult.workflowExecutionId).result.head
+
+          workflowExecutionAuxResult map { workflowExecutionAux =>
+            new WorkflowInfo(
+              UUID.fromString(workflowExecutionResult.workflowExecutionUuid),
+              workflowExecutionAux.wdlSource,
+              workflowExecutionAux.jsonInputs)
+          }
+        }
+      )
+
+    } yield workflowInfos
+
+    runTransaction(action)
+  }
+
+  override def updateWorkflowState(workflowId: WorkflowId, workflowState: WorkflowState): Future[Unit] = {
+    val query = for {
+      workflowExecution <- dataAccess.workflowExecutions
+      if workflowExecution.workflowExecutionUuid === workflowId.toString
+    } yield workflowExecution
+
+    val action = for {
+      count <- query.map(_.status).update(workflowState.toString)
+      _ = require(count == 1, s"Unexpected workflow execution update count $count")
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  override def getExecutionBackendInfo(workflowId: WorkflowId, call: Call): Future[CallBackendInfo] = {
+
+    val action = for {
+
+      workflowExecutionResult <- dataAccess.workflowExecutions.filter(
+        _.workflowExecutionUuid === workflowId.toString).result.head
```

Comment:
Same questions as above. Instead of `Failure`, should we change the method to return `Future[Option[_]]`?